components/form-fields/dynamic-checkbox-group.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class DynamicCheckboxGroup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: [],
+      selected: []
+    };
+  }
+
+  componentDidMount() {
```

Comment:
why does this need to happen when the component mounts, rather than building the options and passing them as `<DynamicCheckboxGroup  options={options} ... />` in the parent?